substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/image/NativeImageDebugInfoProvider.java

```diff
@@ -1057,11 +1106,26 @@ public String methodName() {
             return name;
         }
 
+        @Override
+        public String valueType() {
+            return method.format("%R");
+        }
+
+        @Override
+        public String paramSignature() {
+            return method.format("%P");
+        }
+
         @Override
         public String symbolNameForMethod() {
             return NativeImage.localSymbolNameForMethod(method);
         }
 
+        @Override
+        public boolean isDeoptTarget() {
+            return name().endsWith(HostedMethod.METHOD_NAME_DEOPT_SUFFIX);
```

Comment:
How about `hostedMethod.compilationInfo.isDeoptTarget()` instead?

Reply:
@adinn this was copied from your implementation, do you have any objections on changing it to `hostedMethod.isDeoptTarget()`?

Reply:
> do you have any objections on changing it to hostedMethod.isDeoptTarget()

Not at all. I was not aware that such a method existed when I coded my version. It is far better to use the API method call than rely on the internal details of how deopt method names are encoded.